plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/procedure/FlushMetadataCacheProcedure.java

```diff
@@ -106,13 +104,16 @@ private void doFlushMetadataCache(ConnectorSession session, Optional<String> sch
         }
         else if (schemaName.isPresent() && tableName.isPresent()) {
             HiveMetastore metastore = metastoreFactory.createMetastore(Optional.of(session.getIdentity()));
-            Table table = metastore.getTable(schemaName.get(), tableName.get())
-                            .orElseThrow(() -> new TableNotFoundException(new SchemaTableName(schemaName.get(), tableName.get())));
-            verifyDeltaLakeTable(table);
-            cachingHiveMetastore.ifPresent(caching -> caching.invalidateTable(table.getDatabaseName(), table.getTableName()));
-            String tableLocation = getTableLocation(table);
-            transactionLogAccess.invalidateCaches(tableLocation);
-            extendedStatisticsAccess.invalidateCache(tableLocation);
+            // This may insert into a cache, but this will get invalidated below. TODO fix Delta so that flush_metadata_cache doesn't have to read from metastore
+            Optional<Table> tableBeforeFlush = metastore.getTable(schemaName.get(), tableName.get());
+            cachingHiveMetastore.ifPresent(caching -> caching.invalidateTable(schemaName.get(), tableName.get()));
```

Comment:
Does `invalidateTable` throw something if you call it with a schema/table that isn't cached? Seems like we should be able to just call it blindly without calling `getTable` first.

Reply:
> Does invalidateTable throw something if you call it with a schema/table that isn't cached?

it does not

> Seems like we should be able to just call it blindly without calling getTable first.


we call getTable to know the location (pre-existing)

we invalidate after calling, so that the cache is empty in the end state.